aea/protocols/transaction/message.py

```diff
@@ -0,0 +1,139 @@
+# -*- coding: utf-8 -*-
+
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2019 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""The transaction message module."""
+
+from typing import Dict, cast
+
+from aea.protocols.base import Message
+
+TransactionId = str
+Address = str
+
+
+class TransactionMessage(Message):
+    """The transaction message class."""
+
+    def __init__(self, transaction_id: TransactionId,
+                 sender: Address,
+                 counterparty: Address,
+                 is_sender_buyer: bool,
+                 currency: str,
+                 amount: float,
+                 sender_tx_fee: float,
+                 counterparty_tx_fee: float,
+                 quantities_by_good_pbk: Dict[str, int],
+                 **kwargs):
+        """
+        Instantiate transaction message.
+
+        :param transaction_id: the id of the transaction.
+        :param sender: the sender of the transaction.
+        :param counterparty: the counterparty of the transaction.
+        :param is_sender_buyer: whether the transaction is sent by a buyer.
+        :param currency_pbk: the currency of the transaction.
+        :param sender_tx_fee: the part of the tx fee paid by the sender
+        :param counterparty_tx_fee: the part of the tx fee paid by the counterparty
+        :param amount: the amount of money involved.
+        :param quantities_by_good_pbk: a map from good pbk to the quantity of that good involved in the transaction.
+        """
+        super().__init__(transaction_id=transaction_id,
+                         sender=sender,
+                         counterparty=counterparty,
+                         is_sender_buyer=is_sender_buyer,
+                         currency_pbk=currency,
+                         sender_tx_fee=sender_tx_fee,
+                         counterparty_tx_fee=counterparty_tx_fee,
+                         amount=amount,
+                         quantities_by_good_pbk=quantities_by_good_pbk,
+                         **kwargs)
+        assert self.check_consistency(), "FIPAMessage initialization inconsistent."
```

Comment:
This could be simplified to be:
buyer_pbk
seller_pbk
currency_pbk
seller_tx_fee (share of)
buyer_tx_fee (share of)
amount
quantities_by_good_pbk